presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/api/PackageAPI.java

```diff
@@ -36,87 +35,100 @@
+    @Produces({TEXT_PLAIN})
     @ApiOperation(value = "Install presto using rpm or tarball")
     @ApiResponses(value = {
             @ApiResponse(code = 202, message = "Acknowledged request"),
-            @ApiResponse(code = 400, message = "Bad package format or Invalid url")
+            @ApiResponse(code = 400, message = "Invalid url"),
+            @ApiResponse(code = 409, message = "Presto is already installed.")
     })
-    public synchronized Response install(@NotNull @ApiParam("Url to fetch package") String locationToFetchPackage,
+    public synchronized Response install(@ApiParam("Url to fetch package") String locationToFetchPackage,
             @QueryParam("checkDependencies") @DefaultValue("true") @ApiParam("If false, disables dependency checking") boolean checkDependencies)
     {
         LOGGER.debug("PUT /package  url: {} ; checkDependencies: {}", locationToFetchPackage, checkDependencies);
-        try {
-            installedPackageType = PackageType.valueOf(getFileExtension(locationToFetchPackage).toUpperCase());
-        }
-        catch (IllegalArgumentException e) {
-            LOGGER.error("Invalid package: {}. Package must be .tar.gz or .rpm", locationToFetchPackage, e);
-            return Response.status(BAD_REQUEST).entity(text("Invalid package type for presto-server. Package must be .tar.gz or .rpm")).build();
+        if ("".equals(locationToFetchPackage)) {
+            LOGGER.error("Url is empty or null");
+            return Response.status(BAD_REQUEST).entity("Expected URL in the body").build();
         }
         try {
+            if (isInstalled()) {
+                LOGGER.error("Presto is already installed.");
+                return Response.status(CONFLICT).entity("Presto is already installed.").build();
+            }
             URL url = new URL(locationToFetchPackage);
-            new Thread(new PrestoAsynchronousCommand(new PrestoInstaller(installedPackageType, url, checkDependencies))).start();
+            new Thread(new PrestoAsynchronousCommand(new PrestoInstaller(RPM, url, checkDependencies))).start();
+        }
+        catch (PrestoManagerException e) {
+            LOGGER.error("Failed to ascertain whether presto is running", e);
+            return Response.status(INTERNAL_SERVER_ERROR).entity("Failed to ascertain whether presto is already installed.").build();
         }
         catch (MalformedURLException e) {
             LOGGER.error("Invalid url: {}", locationToFetchPackage, e);
-            return Response.status(BAD_REQUEST).entity(text("Invalid url")).build();
+            return Response.status(BAD_REQUEST).entity("Invalid url").build();
         }
-        return Response.status(ACCEPTED).entity(text("Check status to make sure that presto is installed")).build();
+        return Response.status(ACCEPTED).entity("Presto is being installed. " +
+                "Please check back later using status api to make sure that install is successful.").build();
```

Comment:
1. say "using *the* status api" instead of "using status api"   
2. say "to make sure that installation succeeded"  not "to make sure that install is successful.